go-controller/cmd/ovnkube/ovnkube.go

```diff
@@ -304,7 +394,10 @@ func runOvnKube(ctx *cli.Context, cancel context.CancelFunc) error {
 		if config.Metrics.ExportOVSMetrics {
 			metrics.RegisterOvsMetricsWithOvnMetrics(stopChan)
 		}
-		metrics.RegisterOvnMetrics(ovnClientset.KubeClient, node, stopChan)
+
+		if runMode.node {
+			metrics.RegisterOvnMetrics(ovnClientset.KubeClient, node, stopChan)
+		}
```

Comment:
This is how its presently done - https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/cmd/ovnkube/ovnkube.go#L307

I think this code will not be hit for master mode right   ?
Because of condition

if config.OvnKubeNode.Mode != types.NodeModeDPUHost && config.Metrics.OVNMetricsBindAddress != "" {

Reply:
I guess that the main reason would be `config.Metrics.OVNMetricsBindAddress != ""`

But if that's the case, checking `runMode.node` is superfluous and a bit confusing? Why would we not add that check to the top `if`?

@martinkennelly any thoughts?

Reply:
I 'll just revert this change.  i,e I'll remove the - if runMode.node {